src/Elasticsearch.Net.Aws/Elasticsearch.Net.Aws/AwsHttpConnection.cs

```diff
@@ -9,35 +10,15 @@ namespace Elasticsearch.Net.Aws
     /// </summary>
     public class AwsHttpConnection : HttpConnection
     {
-        private readonly ICredentialsProvider _credentialsProvider;
+        private readonly ImmutableCredentials _credentials;
         private readonly string _region;
 
-        /// <summary>
-        /// Initializes a new instance of the AwsHttpConnection class with the specified AccessKey, SecretKey and Token.
-        /// </summary>
-        /// <param name="awsSettings">AWS specific settings required for signing requests.</param>
-        [Obsolete("Use AwsHttpConnection(string region, ICredentialsProvider credentialsProvider)")]
```

Comment:
Because the rest of this is a major version change, I took the opportunity to remove the obsolete constructor.